pacai/agents/search/base.py

```diff
@@ -16,18 +22,27 @@ class SearchAgent(BaseAgent):
     """
 
     def __init__(self, index,
-            fn = 'pacai.student.search.depthFirstSearch',
-            prob = 'pacai.core.search.position.PositionSearchProblem',
-            heuristic = 'pacai.core.search.heuristic.null',
+            fn: Union[str, Callable[[SearchProblem], any]] = depthFirstSearch,
+            prob: Union[str, Callable[[AbstractGameState], SearchProblem]] = PositionSearchProblem,
+            heuristic: str = 'pacai.core.search.heuristic.null',
```

Comment:
Give `heuristic` the same treatment as `fn` and `prob`.
The conversion from string to function can happen inside `_fetchSearchFunction`, and the check against `heuristic` on line 39 can probably just be skipped.